hexa/workspaces/middlewares.py

```diff
@@ -0,0 +1,38 @@
+import binascii
+from logging import getLogger
+
+from django.core.signing import BadSignature, Signer
+from django.http import HttpRequest
+
+from hexa.workspaces.models import WorkspaceMembership
+
+logger = getLogger(__name__)
+
+
+# FIXME: This needs to be changed as it allows the user to query everything with just a simple workspace token.
+#        We need to restrict the user's actions to only the workspace he is part of.
+def workspace_authentication_middleware(get_response):
+    """This middleware allows an user to be authenticated through a simple token linked to a workspace"""
+
+    def middleware(request: HttpRequest):
+        try:
+            auth_type, token = request.headers["Authorization"].split(" ")
+            if auth_type.lower() == "bearer":
+                token = Signer().unsign_object(token)
+                membership = WorkspaceMembership.objects.get(access_token=token)
+                request.user = membership.user
```

Comment:
I think we should create a dedicated user class for workspace tokens. This user class would be linked to the user and workspace, and would respect the Django user interface.

This would ease the differentiation between logged-in users and CLI users when it matters.

It would require a bit of work in our workspace permission checks but I think it would be more robust than setting `workspace` and `bypass_two_factor` on a regular user.

Reply:
Actually the user class might not be a good idea - it makes sense for a pipeline run but in this case it's really the user that is performing operations.

What should be changed IMHO is to set the workspace / bypass params on the request and not on the user.